src/veh_interact.cpp

```diff
@@ -454,8 +469,7 @@ task_reason veh_interact::cant_do (char mode)
     case 'c': // change tire
         valid_target = wheel != NULL;
         ///\EFFECT_STR allows changing tires on heavier vehicles without a jack
-        has_str = g->u.get_str() >= int(veh->total_mass() / TIRE_CHANGE_STR_MOD);
-        has_tools = has_wrench && (has_jack || has_str) && has_wheel;
+        has_tools = has_wrench && ( g->u.can_lift( *veh ) || has_jack ) && has_wheel;
```

Comment:
Swapping the `bool` and function invocation around would skip a potentially expensive invocation.